generative_ai/distillation.py

```diff
@@ -23,37 +25,37 @@
 from vertexai.preview.language_models import TextGenerationModel, TuningEvaluationSpec
 
 
-credentials, _ = default(scopes=["https://www.googleapis.com/auth/cloud-platform"])
+CREDENTIALS, _ = default(scopes=["https://www.googleapis.com/auth/cloud-platform"])
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+LOCATION = "us-central1"
 
 
 def distill_model(
-    project_id: str,
-    location: str,
     dataset: str,
     teacher_model: str,
-    train_steps: int = 300,
     evaluation_dataset: Optional[str] = None,
 ) -> None:
-    """Distill a new model.
-
+    """Distill a new model using a teacher model and a dataset.
     Args:
-      project_id: GCP Project ID, used to initialize vertexai
-      location: GCP Region, used to initialize vertexai
-      dataset: GCS URI of jsonl file.
-      teacher_model: Name of the teacher model.
-      train_steps: Number of training steps to use when tuning the model.
-      evaluation_dataset: GCS URI of jsonl file of evaluation data.
+        dataset (str): GCS URI of the JSONL file containing the training data.
+        teacher_model (str): Name of the teacher model to distill from.
+        evaluation_dataset (Optional[str]): GCS URI of the JSONL file containing the evaluation data.
     """
-    vertexai.init(project=project_id, location=location, credentials=credentials)
+    # TODO developer - override these parameters as needed:
+    vertexai.init(project=PROJECT_ID, location=LOCATION, credentials=CREDENTIALS)
```

Comment:
Is passing the credentials explicitly like this required? Doesn't the `vertexai.init` function look for the default credentials if none is passed?

Reply:
I think, it should work without explicitly passing `credentials` 👍